python/pyarrow/_dataset_parquet.pyx

```diff
@@ -127,8 +127,14 @@ cdef class ParquetFileFormat(FileFormat):
                             'instance of ParquetReadOptions')
 
         if default_fragment_scan_options is None:
-            default_fragment_scan_options = ParquetFragmentScanOptions(
-                **scan_args)
+            # remove decryption_properties from scan_args as it does not take this parameter
+            decryption_properties = scan_args.pop('decryption_properties', None)
+            default_fragment_scan_options = ParquetFragmentScanOptions(**scan_args)
+            # make sure scan options has decryption properties
+            if decryption_properties is not None:
+                default_fragment_scan_options.set_file_decryption_properties(
+                    decryption_properties)
+
         elif isinstance(default_fragment_scan_options, dict):
```

Comment:
Should we also make sure that `decryption_properties` are set if `default_fragment_scan_options` are specified as a dictionary together with  `decryption_properties`?

Though, if the dictionary is used to pass options to `ParquetFileFormat` it will be considered as `**kwargs` so I am not even sure if there is any use case to support `default_fragment_scan_options` being a dictionary?

Also, would it be better to add `decryption_properties` to `ParquetFragmentScanOptions` `reader_properties()`? This way the logic for removing `decryption_properties` from `scan_args` and setting them manually via `set_file_decryption_properties` would not be needed?

Reply:
Good point. Refactored.